tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/cli/BulkCommittingCommand.java

```diff
@@ -0,0 +1,251 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.tools.contentgenerator.cli;
+
+import static java.util.Collections.singletonList;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Iterators;
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import org.projectnessie.client.api.NessieApiV2;
+import org.projectnessie.error.BaseNessieClientServerException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.EntriesResponse;
+import org.projectnessie.model.Reference;
+import picocli.CommandLine.Option;
+
+public abstract class BulkCommittingCommand extends CommittingCommand {
+
+  @Option(
+      names = {"--all"},
+      description =
+          "Process all active keys in all branches (supersedes --input and all other selector options).")
+  private boolean all;
+
+  @Option(
+      names = {"-r", "--ref", "--branch"},
+      description = "Branch name for making changes (default branch if not set).")
+  private String ref;
+
+  @Option(
+      names = {"-f", "--input"},
+      description = "Input file name. See the --format option for details (if not set, use --key).")
+  private String input;
+
+  @Option(
+      names = {"-F", "--format"},
+      description =
+          "The format of the input file. CSV_KEYS means one content key per line (separated key elements or "
+              + "URL path encoded whole key). CONTENT_INFO_JSON means a JSON array of 'ContentInfoEntry' objects from "
+              + "the 'content-info' CLI command.")
+  private InputFormat format;
+
+  @Option(
+      names = {"-S", "--separator"},
+      description =
+          "The fields separator for CVS input files (if not set, each line is interpreted as a URL path "
+              + "encoded key string).")
+  private String separator;
+
+  @Option(
+      names = {"--storage-model"},
+      description =
+          "If set, only those entries that have the specified storage model will be processed. "
+              + "This option is applicable only to JSON input files. (ignored if --input is not set).")
+  private String storageModel;
+
+  @Option(
+      names = {"--skip-tags"},
+      description =
+          "If set, input references that are not branches will be ignored (as opposed to reported as errors).")
+  private boolean skipTags;
```

Comment:
Not sure whether this flag's actually useful. If not set, the commands will probably fail on all repos, potentially after processing a bunch of branches.
I think, the commands should only fail, if a reference was explicitly specified and that's a tag.

Reply:
removed this option